### PR TITLE
bugfix:the buffer may cause overflows when sql statement is long

### PR DESCRIPTION
**Pull request by kezhenxu94**
### Ⅰ. Describe what this PR did
Due to the `java.nio.ByteBuffer` can not grow dynamically, its size is hardcoded now, this patch replaces `java.nio.ByteBuffer` with `io.netty.buffer.ByteBuf`, which can grow on demand.

### Ⅱ. Does this pull request fix one issue?
fixes issue #976 

### Ⅲ. Why don't you add test cases (unit test/integration test)? 
There is already tests upon this method.

### Ⅳ. Describe how to verify it
CI passes

### Ⅴ. Special notes for reviews
null

### CODE REVIEW

File: core/src/main/java/io/seata/core/protocol/MergedWarpMessage.java [hunk 1]
```diff
@@ -49,21 +51,21 @@ public short getTypeCode() {
...
     public byte[] encode() {
-        int bufferSize = msgs.size() * 1024;
-        ByteBuffer byteBuffer = ByteBuffer.allocate(bufferSize);
-        byteBuffer.putShort((short)msgs.size());
-        for (AbstractMessage msg : msgs) {
-            //msg.setChannelHandlerContext(ctx);
-            byte[] data = msg.encode();
-            byteBuffer.putShort(msg.getTypeCode());
-            byteBuffer.put(data);
+        final ByteBuf buffer = Unpooled.buffer();
```
Comment (leizhiyuan) [inline core/src/main/java/io/seata/core/protocol/MergedWarpMessage.java]: why do you use unpooled? maybe pooled is safer?

Comment (kezhenxu94) [reply core/src/main/java/io/seata/core/protocol/MergedWarpMessage.java]: make sense, but it's what Netty recommends to create a `ByteBuf`:

https://github.com/netty/netty/blob/128403b492d8490d9fc770b7a2dad669433b2790/buffer/src/main/java/io/netty/buffer/ByteBuf.java#L37-L41

and there is no something like `Pooled`, or should I call `new PooledByteBufAllocator(false).buffer()` directly? which is not recommended.

What do you think? @leizhiyuan

Comment (slievrly) [inline core/src/main/java/io/seata/core/protocol/MergedWarpMessage.java]: default buffer size=256，can we set a larger size and reduce the number of resize?

Comment (kezhenxu94) [reply core/src/main/java/io/seata/core/protocol/MergedWarpMessage.java]: @slievrly maybe set to what we hardcoded before? (say 1024)

**Pull request merged**